cfme/common/provider.py

```diff
@@ -119,9 +119,6 @@ def __hash__(self):
     def __eq__(self, other):
         return type(self) is type(other) and self.key == other.key
 
-    def __str__(self):
```

Comment:
Didn't you submit a PR to put this in? Should this be here?

Reply:
Yes. I removed it from here before the other PR was merged and rebase removed it from master. I am adding it back and also adding back it's usage here.

Reply:
Yeah. Rebase removed it from master. I added the commit that removed it before it was merged. We wanted it in it's own PR so additional testing could be done. I'll add it back in and also reimplement the usage of it.